validation/validate.py

```diff
@@ -19,17 +26,29 @@ def validate(collection_path):
             try:
                 in_data = yaml.safe_load(stream)
             except yaml.YAMLError as exc:
-                print(exc)
-
-        # validate it
-        # try:
-        #     CollectionSchema().load(in_data)
-        # except ValidationError as err:
-        #     pprint(err.messages)
-
-        print(f"validating {collection}")
-        parsed = CollectionSchema.parse_obj(in_data)
-        print(parsed)
+                yaml_error_list.append((str(exc.problem) + str(exc.problem_mark)))
+                continue
+
+        # process the Collection
+        try:
+            CollectionSchema.parse_obj(in_data)
+        except ValidationError as err:
+            schema_error_list.append(str(err) + ' in ' + collection)
+
+    if len(yaml_error_list) == 0 and len(schema_error_list) == 0:
```

Comment:
edge case to consider:

either file loading OR deserialization throws an error BUT its not a `YAMLError` or a `ValidationError`...

what happens?

Reply:
In that case, I believe we need to add a general error case to catch them!

Reply:
I think we are unlikely to run into a FileNotFound Error, as the file names are fetched from glob. As for running into deserialization errors, if the YAML parser encounters an error condition, it raises an exception which is an instance of YAMLError or of its subclass, which should handle the deserialization bit. Therefore, the current two edge cases should capture all of the use cases.